### PR TITLE
Create Google Oauth Config variable to overwrite settings.py

### PR DESCRIPTION
**Pull request by RynhardPietersen**
[link](https://github.com/dimagi/commcare-hq/pull/30982) to pr which prompted this change

I'm simply adding some secrets for use in localsettings.py in order to overwrite the client secrets dict in settings.py

### CODE REVIEW

**Comment by RynhardPietersen** [comment]
I've added the 'do not merge' label because I'm not entirely sure if this has been done correctly

File: src/commcare_cloud/environment/secrets/secrets.yml [hunk 1]
```diff
@@ -77,6 +77,15 @@
   legacy_namespace: localsettings_private
   default: ''
   default_overrides_falsy_values: yes
+- name: GOOGLE_OAUTH_CLIENT_ID
```
Comment (biyeun) [inline src/commcare_cloud/environment/secrets/secrets.yml:80]: @dannyroberts is this all that's needed to add new secrets?

**Review by biyeun — APPROVED** [review]

File: src/commcare_cloud/environment/secrets/secrets.yml (hunk 1 above)
Comment (dannyroberts) [reply src/commcare_cloud/environment/secrets/secrets.yml:80]: Here are the most up to date instructions for adding new secrets: https://confluence.dimagi.com/display/saas/How+to+add+new+secrets+or+other+local+settings. I see there's also the step of updating tests accordingly for example. But yeah following that should be what you need.

File: src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2 [hunk 2]
```diff
@@ -342,6 +342,17 @@ ANALYTICS_IDS = {
...
+    "web": {
+        "client_id": '{{ GOOGLE_OAUTH_CLIENT_ID }}',
+        "project_id": '{{ GOOGLE_OAUTH_PROJECT_ID }}',
+        "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+        "token_uri": "https://oauth2.googleapis.com/token",
+        "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+        "client_secret": '{{ GOOGLE_OAUTH_CLIENT_SECRET }}'
+    }
+}
+
```
Comment (mjriley) [inline src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2:355]: Is google sheets functionality optional for self-hosters? If so, we might think about putting this block in a conditional, depending on if the google oauth values are present

**Review by mjriley — APPROVED** [review]

File: src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2 (hunk 2 above)
Comment (RynhardPietersen) [reply src/commcare_cloud/ansible/roles/commcarehq/templates/localsettings.py.j2:355]: we haven't really spoken about self hosters yet

**Pull request merged**